powerstore/resource_volume_group.go

```diff
@@ -236,6 +236,144 @@ func (r *resourceVolumeGroup) Read(ctx context.Context, req resource.ReadRequest
 
 // Update - method to update volume group resource
 func (r *resourceVolumeGroup) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	log.Printf("Started Update")
+
+	//Get plan values
+	var plan models.Volumegroup
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	//Get current state
+	var state models.Volumegroup
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var planVolumeIds []string
+	for _, volume := range plan.VolumeIDs.Elements() {
+		planVolumeIds = append(planVolumeIds, strings.Trim(volume.String(), "\""))
+	}
+
+	var stateVolumeIds []string
+	for _, volume := range state.VolumeIDs.Elements() {
+		stateVolumeIds = append(stateVolumeIds, strings.Trim(volume.String(), "\""))
+	}
+
+	planVolumeIdsMap := make(map[string]string)
+	if len(planVolumeIds) != 0 {
+		for i := 0; i < len(planVolumeIds); i++ {
+			planVolumeIdsMap[planVolumeIds[i]] = planVolumeIds[i]
+		}
+	}
+
+	stateVolumeIdsMap := make(map[string]string)
+	if len(stateVolumeIds) != 0 {
+		for i := 0; i < len(stateVolumeIds); i++ {
+			stateVolumeIdsMap[stateVolumeIds[i]] = stateVolumeIds[i]
+		}
+	}
+
+	removeVolumeIdsMap := make(map[string]string)
+	for i := 0; i < len(stateVolumeIds); i++ {
+		_, found := planVolumeIdsMap[stateVolumeIds[i]]
+		if !found {
+			log.Printf("Elem not found in state")
+			removeVolumeIdsMap[stateVolumeIds[i]] = stateVolumeIds[i]
+		}
+	}
+
+	removeVolumeIdsSlice := []string{}
```

Comment:
Please add the comments for readability purposes. Applicable for all such instances.

Reply:
done